src/contents/l1sload-guide-read-arrays-mappings-structs.mdx

```diff
@@ -0,0 +1,422 @@
+        if(!success)
+        {
+            revert("L1SLOAD failed");
+        }
+        return abi.decode(returnValue, (uint));
+    }
+
+    function getArrayLength(address contractAddress, uint arraySlot) public view returns(uint) {
+        bool success;
+        bytes memory returnValue;
+        (success, returnValue) = L1_SLOAD_ADDRESS.staticcall(abi.encodePacked(contractAddress, arraySlot));
+        if(!success)
+        {
+            revert("L1SLOAD failed");
+        }
+        return abi.decode(returnValue, (uint));
+    }
+}
+```
+
+## Special Case: Storing Values 16 Bytes or Smaller
+
+Consider a dynamic array of `uint104` values. Since two `uint104` values can fit into a single 32-byte storage slot (with 48 bits, or 6 bytes, remaining unused), Solidity packs them together.
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract L1Uint104Array {
+    uint104[] public myArray;
+
+	constructor() {
+		myArray.push(10);
+		myArray.push(20);
+		myArray.push(30);
+		myArray.push(40);
+		myArray.push(50);
+	}
+}
+```
+
+<div className="text-center">
+  <img
+    src="https://dev-to-uploads.s3.amazonaws.com/uploads/articles/n80x3swtgbvrdpvpxzwk.png"
+    alt="Uint104 Array Storage Demo"
+    width="950"
+  />
+  _L1Uint104Array storage layout_
+</div>
+
+To query such values using `l1sload`, we first locate the appropriate slot, then use bitwise operations to extract the desired value. The following formula helps in calculating the slot:
+
+
+<div className="text-center">
+  <img
+    src="https://dev-to-uploads.s3.amazonaws.com/uploads/articles/ylfzimoadebv1ihcgh0l.png"
+    alt="Formula Dynamic Arrays with other sizes getting the slot"/>
+</div>
+
+However, retrieving the value also requires shifting the bits appropriately, which can be done using the right-shift operator.
+
+<div className="text-center">
+  <img
+    src="https://dev-to-uploads.s3.amazonaws.com/uploads/articles/ox8ggvx31yxgxx9bmrpp.png"
+    alt="Shifting bytes to get value once we have the slot"/>
+</div>
+
+This method can also work for larger types like uint256 or address, but for simplicity and efficiency, I recommend sticking to the methods mentioned earlier for those types.
+
+Here’s an example contract that queries data stored in a uint104 array:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract L1Uint104ArrayDemo {
+    address constant L1_BLOCKS_ADDRESS = 0x5300000000000000000000000000000000000001;
+    address constant L1_SLOAD_ADDRESS = 0x0000000000000000000000000000000000000101;
+
+    function retrieveArrayElement(address contractAddress, uint arraySlot, uint arrayIndex) public view returns(uint104) {
+        require(arrayIndex < getArrayLength(contractAddress, arraySlot), "Out of bounds index");
+        
+		uint arrayElementSlot = uint(
+            keccak256(
+                abi.encodePacked(arraySlot)
+            )
+        ) + arrayIndex / (uint(245)/104);
```

Comment:
@Turupawn just noted one more item for clarification in my second-pass review - do you know why we use slot size of `245` here?

Reply:
@PrintRH Fixed!